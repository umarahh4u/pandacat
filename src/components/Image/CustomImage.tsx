import NextImage from "next/image";
import { chakra, Box } from "@chakra-ui/react";

const Image = chakra(NextImage, {
  // baseStyle: { maxH: 1120, maxW: 1120 },
  shouldForwardProp: (prop) =>
    [
      "width",
      "height",
      "src",
      "alt",
      "placeholder",
      "blurDataURL",
      "_hover",
      "layout",
    ].includes(prop),
});

const CustomImage = (props: any) => {
  const { src, width, height, alt, layout, placeholder, blurDataURL, ...rest } =
    props;
  // const defaultWidth = width ? width : ''

  return (
    <Box position="relative" {...rest}>
      <Image
        src={src}
        alt={alt}
        width={width}
        height={height}
        layout={layout}
        placeholder="blur"
        blurDataURL="/blur.jpeg"
        transition="all 0.5s"
      />
    </Box>
  );
};

export default CustomImage;
