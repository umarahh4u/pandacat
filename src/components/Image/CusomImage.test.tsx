import { render } from '@testing-library/react'
import CustomImage from '.'

it('renders CustomImage component', () => {
  const { container } = render(
    <CustomImage src="/apple-touch-icon.png" width={50} height={50} />
  )
  expect(container).toMatchSnapshot()
})
