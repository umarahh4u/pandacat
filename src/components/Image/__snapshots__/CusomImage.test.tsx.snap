// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomImage component 1`] = `
<div>
  <div
    class="css-79elbk"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        class="css-bmhb71"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; background-size: cover; background-position: 0% 0%; filter: blur(20px); background-image: url(/blur.jpeg);"
      />
      <noscript />
    </span>
  </div>
</div>
`;
